src/giskardpy/plugin_instantaneous_controller.py

```diff
@@ -43,7 +49,36 @@ def init_controller(self):
             self.soft_constraints = copy(new_soft_constraints)
             self.controller = InstantaneousController(self.get_robot(),
                                                   u'{}/{}/'.format(self.path_to_functions, self.get_robot().get_name()))
-            self.controller.set_controlled_joints(self.get_robot().controlled_joints)
+
+            controlled_joints = self.get_robot().controlled_joints
+            joint_to_symbols_str = OrderedDict(
+                (x, self.robot.get_joint_position_symbol(x)) for x in controlled_joints)
+
+            #make_filter_masks(self.get_god_map().safe_get_data(identifier.H), )
+
+            if(self.get_god_map().safe_get_data(identifier.check_reachability) and False):
+                joint_constraints = OrderedDict()
+                for k in controlled_joints:
+                    joint_type = self.get_robot().get_joint_type(k)
+                    weight = self.robot._joint_constraints[k].weight
+                    if joint_type == 'prismatic':
+                        joint_constraints[(self.robot.get_name(), k)] = JointConstraint(-self.rc_prismatic_velocity, self.rc_prismatic_velocity, weight)
+                    elif joint_type == 'continuous':
+                        joint_constraints[(self.robot.get_name(), k)] = JointConstraint(-self.rc_continuous_velocity, self.rc_continuous_velocity, weight)
+                    elif joint_type == 'revolute':
+                        joint_constraints[(self.robot.get_name(), k)] = JointConstraint(-self.rc_revolute_velocity, self.rc_revolute_velocity, weight)
+                    else:
+                        joint_constraints[(self.robot.get_name(), k)] = JointConstraint(-self.rc_other_velocity, self.rc_other_velocity, weight)
+            else:
+                joint_constraints = OrderedDict(((self.robot.get_name(), k), self.robot._joint_constraints[k]) for k in
+                                                     controlled_joints)
+            hard_constraints = OrderedDict(((self.robot.get_name(), k), self.robot._hard_constraints[k]) for k in
+                                                controlled_joints if k in self.robot._hard_constraints)
```

Comment:
In `plugin_update_constraints.py` there is `GoalToConstraints`. It probably makes sense to move this code to there, please do that.